sea-orm-cli/src/commands.rs

```diff
@@ -150,7 +151,9 @@ pub async fn run_generate_command(matches: &ArgMatches<'_>) -> Result<(), Box<dy
                 _ => unimplemented!("{} is not supported", url.scheme()),
             };
 
-            let output = EntityTransformer::transform(table_stmts)?
+            let name_resolver = NameResolver::new(singularize);
+
+            let output = EntityTransformer::transform(table_stmts, name_resolver)?
                 .generate(expanded_format, WithSerde::from_str(with_serde).unwrap());
```

Comment:
Hey @negezor, thanks for the updates! I think the semantic isn't seems right. The transformer, `EntityTransformer::transform(table_stmts, name_resolver)`, don't need name resolver. Instead, we could have introduce a new struct call `EntityWriterContext` which contains three things...

```rs
pub struct EntityWriterContext {
  pub(crate) expanded_format: bool,
  pub(crate) with_serde: WithSerde,
  pub(crate) name_resolver: NameResolver,
}
```

Then, `EntityWriter::generate` method would take an `EntityWriterContext`. Thoughts?

Reply:
I agree indeed

Reply:
@billy1624 In the case where we discussed about saving a context info on which datetime crate the user chooses in [the `--date-time-crate` flag for `sea-orm-cli`](https://github.com/SeaQL/sea-orm/issues/661), we'd basically write:
```rust
pub struct EntityWriterContext {
  pub(crate) expanded_format: bool,
  pub(crate) with_serde: WithSerde,
  pub(crate) name_resolver: NameResolver,
  pub(crate) date_time_crate: &str,
}
```
and pass this context all the way down to `Column.get_rs_type()` for resolving the Rust types based on `chrono` vs `time`? :)